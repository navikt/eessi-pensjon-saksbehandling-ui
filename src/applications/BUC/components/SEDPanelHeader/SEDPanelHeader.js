import React from 'react'
import { Nav } from 'eessi-pensjon-ui'
import './SEDPanelHeader.css'

const SEDPanelHeader = ({ t }) => (
  <div className='a-buc-c-sedpanelheader mt-2 mb-2'>
    <div className='a-buc-c-sedpanelheader_head col-4'>
      <Nav.Element>{t('buc:form-sed')}</Nav.Element>
    </div>
    <div className='a-buc-c-sedpanelheader_head col-3'>
      <Nav.Element>{t('buc:form-sender')}</Nav.Element>
    </div>
    <div className='a-buc-c-sedpanelheader_head col-3'>
      <Nav.Element>{t('buc:form-receiver')}</Nav.Element>
    </div>
    <div className='a-buc-c-sedpanelheader_head col-2'>
      <Nav.Element>{t('buc:form-actions')}</Nav.Element>
    </div>
  </div>
)

export default SEDPanelHeader
