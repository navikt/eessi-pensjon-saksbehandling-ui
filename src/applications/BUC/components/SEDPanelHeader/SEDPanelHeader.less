@import '../../../../constants';

.a-buc-c-sedpanelheader {
  display: flex;
  justify-content: space-between;
  border-bottom: 3px solid @blue-nav;
}
