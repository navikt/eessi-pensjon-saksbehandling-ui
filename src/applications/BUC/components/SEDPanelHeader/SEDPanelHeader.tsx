import { Element } from 'nav-frontend-typografi'
import { theme, themeHighContrast } from 'nav-styled-component-theme'
import React from 'react'
import { useTranslation } from 'react-i18next'
import styled, { ThemeProvider } from 'styled-components'

const SEDPanelHeaderDiv = styled.div`
  display: flex;
  justify-content: space-between;
  border-bottom: 3px solid ${({ theme }: any) => theme.navBla};
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
`
const Flex4Div = styled.div`
  flex: 4;
`
const Flex3Div = styled.div`
  flex: 3;
`
const Flex2Div = styled.div`
  flex: 2;
`
export interface SEDPanelHeaderProps {
  highContrast: boolean
}
const SEDPanelHeader: React.FC<SEDPanelHeaderProps> = ({
  highContrast
}: SEDPanelHeaderProps): JSX.Element => {
  const { t } = useTranslation()
  return (
    <ThemeProvider theme={highContrast ? themeHighContrast : theme}>
      <SEDPanelHeaderDiv>
        <Flex4Div>
          <Element>{t('buc:form-sed')}</Element>
        </Flex4Div>
        <Flex3Div>
          <Element>{t('buc:form-sender')}</Element>
        </Flex3Div>
        <Flex3Div>
          <Element>{t('buc:form-receiver')}</Element>
        </Flex3Div>
        <Flex2Div>
          <Element>{t('buc:form-actions')}</Element>
        </Flex2Div>
      </SEDPanelHeaderDiv>
    </ThemeProvider>
  )
}

SEDPanelHeader.propTypes = {}

export default SEDPanelHeader
