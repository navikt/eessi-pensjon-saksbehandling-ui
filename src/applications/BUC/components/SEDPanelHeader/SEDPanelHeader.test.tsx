import { render, screen } from '@testing-library/react'
import SEDPanelHeader from './SEDPanelHeader'

describe('applications/BUC/components/SEDPanelHeader/SEDPanelHeader', () => {
  it('Render: match snapshot', () => {
    const { container } = render(<SEDPanelHeader />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('Render: has proper HTML structure', () => {
    render(<SEDPanelHeader />)
    expect(screen.getByTestId('a_buc_c_SEDPanelHeader--sed')).toBeInTheDocument()
    expect(screen.getByTestId('a_buc_c_SEDPanelHeader--sender')).toBeInTheDocument()
    expect(screen.getByTestId('a_buc_c_SEDPanelHeader--receiver')).toBeInTheDocument()
    expect(screen.getByTestId('a_buc_c_SEDPanelHeader--actions')).toBeInTheDocument()
  })
})
