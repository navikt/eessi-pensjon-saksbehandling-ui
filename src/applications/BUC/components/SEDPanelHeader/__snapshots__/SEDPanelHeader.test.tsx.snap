// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applications/BUC/components/SEDPanelHeader/SEDPanelHeader Render: match snapshot 1`] = `
.c7 {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  width: 100%;
}

ui-fss.g27[id="c6"] {
  content: "c7,";
}

.c3 {
  -webkit-flex: 4;
  -ms-flex: 4;
  flex: 4;
  width: 100%;
}

ui-fss.g28[id="c2"] {
  content: "c3,";
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border-bottom: 3px solid #0067C5;
  margin-top: 1.5rem;
  margin-bottom: 1rem;
  padding-left: 1rem;
  padding-right: 1rem;
}

ui-fss.g29[id="c0"] {
  content: "c1,";
}

.c5 {
  -webkit-flex: 3;
  -ms-flex: 3;
  flex: 3;
  width: 100%;
}

ui-fss.g30[id="c4"] {
  content: "c5,";
}

<SEDPanelHeader
  highContrast={false}
>
  <Se
    highContrast={false}
  >
    <Component
      theme={
        Object {
          "alternative-background-color": "#E9E7E7",
          "black": "#000000",
          "border-radius-base": "4px",
          "border-radius-large": "6px",
          "border-radius-small": "2px",
          "container-desktop": "(940px + @grid-gutter-width)",
          "container-large-desktop": "(1140px + @grid-gutter-width)",
          "container-lg": "(1140px + @grid-gutter-width)",
          "container-md": "(940px + @grid-gutter-width)",
          "container-sm": "(720px + @grid-gutter-width)",
          "container-tablet": "(720px + @grid-gutter-width)",
          "fokusFarge": "#254a6c",
          "grayBackground": "#E9E7E7",
          "grayIcon": "#B7B1A9",
          "grayInactive": "#78706A",
          "grayModia": "#333333",
          "grid-columns": "12",
          "grid-float-breakpoint": "768px",
          "grid-float-breakpoint-max": "(@grid-float-breakpoint - 1)",
          "grid-gutter-width": "16px",
          "inverted-font-color": "#FFFFFF",
          "line-height-large": "1.3333333",
          "line-height-small": "1.5",
          "main-active-color": "#a3dff3",
          "main-background-color": "#FFFFFF",
          "main-border-color": "#78706A",
          "main-border-radius": "4px",
          "main-disabled-color": "#B7B1A9",
          "main-error-color": "#BA3A26",
          "main-focus-color": "#254a6c",
          "main-font-color": "#3E3832",
          "main-hover-color": "#e0f4fb",
          "main-interactive-color": "#0067C5",
          "main-success-color": "#06893A",
          "main-warning-color": "#FF9100",
          "navBla": "#0067C5",
          "navBlaDarken20": "#0c5da7",
          "navBlaDarken40": "#18548a",
          "navBlaDarken60": "#254a6c",
          "navBlaDarken80": "#31414f",
          "navBlaLighten20": "#3385d0",
          "navBlaLighten40": "#66a3dc",
          "navBlaLighten60": "#99c2e7",
          "navBlaLighten80": "#cce0f3",
          "navDypBla": "#005B82",
          "navDypBlaDarken20": "#0c5472",
          "navDypBlaDarken40": "#184d62",
          "navDypBlaDarken60": "#254652",
          "navDypBlaDarken80": "#313f42",
          "navDypBlaLighten20": "#337b9b",
          "navDypBlaLighten40": "#669cb4",
          "navDypBlaLighten60": "#99bdcd",
          "navDypBlaLighten80": "#ccdee6",
          "navGra20": "#C6C2BF",
          "navGra40": "#B7B1A9",
          "navGra60": "#78706A",
          "navGra80": "#59514B",
          "navGronn": "#06893A",
          "navGronnDarken20": "#117838",
          "navGronnDarken40": "#1c6836",
          "navGronnDarken60": "#275835",
          "navGronnDarken80": "#324833",
          "navGronnLighten20": "#37a061",
          "navGronnLighten40": "#69b888",
          "navGronnLighten60": "#9bcfb0",
          "navGronnLighten80": "#cde7d7",
          "navLilla": "#634689",
          "navLillaDarken20": "#5b4377",
          "navLillaDarken40": "#544066",
          "navLillaDarken60": "#4c3d54",
          "navLillaDarken80": "#453a43",
          "navLillaLighten20": "#826ba0",
          "navLillaLighten40": "#a190b8",
          "navLillaLighten60": "#c0b5cf",
          "navLillaLighten80": "#dfdae7",
          "navLimeGronn": "#A2AD00",
          "navLimeGronnDarken20": "#8e950a",
          "navLimeGronnDarken40": "#7a7e14",
          "navLimeGronnDarken60": "#66661e",
          "navLimeGronnDarken80": "#524f28",
          "navLimeGronnLighten20": "#b4bd33",
          "navLimeGronnLighten40": "#c7cd66",
          "navLimeGronnLighten60": "#d9de99",
          "navLimeGronnLighten80": "#eceecc",
          "navLysBla": "#66CBEC",
          "navLysBlaDarken20": "#5eadc6",
          "navLysBlaDarken40": "#5690a1",
          "navLysBlaDarken60": "#4e727c",
          "navLysBlaDarken80": "#465557",
          "navLysBlaLighten20": "#84d5ef",
          "navLysBlaLighten40": "#a3dff3",
          "navLysBlaLighten60": "#c1eaf7",
          "navLysBlaLighten80": "#e0f4fb",
          "navLysGra": "#E9E7E7",
          "navMorkGra": "#3E3832",
          "navOransje": "#FF9100",
          "navOransjeDarken20": "#d87f0a",
          "navOransjeDarken40": "#b16d14",
          "navOransjeDarken60": "#8b5b1e",
          "navOransjeDarken80": "#644928",
          "navOransjeLighten20": "#ffa733",
          "navOransjeLighten40": "#ffbd66",
          "navOransjeLighten60": "#ffd399",
          "navOransjeLighten80": "#ffe9cc",
          "navRod": "#C30000",
          "navRodDarken20": "#a80b0a",
          "navRodDarken40": "#8d1614",
          "navRodDarken60": "#73211e",
          "navRodDarken80": "#582c28",
          "navRodLighten20": "#cf3333",
          "navRodLighten40": "#db6666",
          "navRodLighten60": "#e79999",
          "navRodLighten80": "#f3cccc",
          "orangeFocus": "#FFBD66",
          "orangeFocusDarken20": "#d8a25b",
          "orangeFocusDarken40": "#b18751",
          "orangeFocusDarken60": "#8b6d46",
          "orangeFocusDarken80": "#64523c",
          "orangeFocusLighten20": "#ffca84",
          "orangeFocusLighten40": "#ffd7a3",
          "orangeFocusLighten60": "#ffe4c1",
          "orangeFocusLighten80": "#fff1e0",
          "padding-base-horizontal": "12px",
          "padding-base-vertical": "6px",
          "padding-large-horizontal": "16px",
          "padding-large-vertical": "10px",
          "padding-small-horizontal": "10px",
          "padding-small-vertical": "5px",
          "padding-xs-horizontal": "5px",
          "padding-xs-vertical": "1px",
          "pinkErrorBg": "#F3E3E3",
          "redError": "#BA3A26",
          "redErrorDarken20": "#a13928",
          "redErrorDarken40": "#88392a",
          "redErrorDarken60": "#6f382d",
          "redErrorDarken80": "#56382f",
          "redErrorLighten20": "#c76151",
          "redErrorLighten40": "#d5887c",
          "redErrorLighten60": "#e3b0a8",
          "redErrorLighten80": "#f1d7d3",
          "screen-desktop": "992px",
          "screen-lg": "1200px",
          "screen-lg-desktop": "1200px",
          "screen-lg-min": "1200px",
          "screen-md": "992px",
          "screen-md-max": "(@screen-lg-min - 1)",
          "screen-md-min": "992px",
          "screen-sm": "768px",
          "screen-sm-max": "(@screen-md-min - 1)",
          "screen-sm-min": "768px",
          "screen-tablet": "768px",
          "screen-xs-max": "(@screen-sm-min - 1)",
          "type": "theme",
          "white": "#FFFFFF",
        }
      }
    >
      <styled.div>
        <div
          className="c0 c1"
        >
          <styled.div>
            <div
              className="c2 c3"
            >
              <Element>
                <TypografiBase
                  type="element"
                >
                  <p
                    className="typo-element"
                  >
                    buc:form-sed
                  </p>
                </TypografiBase>
              </Element>
            </div>
          </styled.div>
          <styled.div>
            <div
              className="c4 c5"
            >
              <Element>
                <TypografiBase
                  type="element"
                >
                  <p
                    className="typo-element"
                  >
                    buc:form-sender
                  </p>
                </TypografiBase>
              </Element>
            </div>
          </styled.div>
          <styled.div>
            <div
              className="c4 c5"
            >
              <Element>
                <TypografiBase
                  type="element"
                >
                  <p
                    className="typo-element"
                  >
                    buc:form-receiver
                  </p>
                </TypografiBase>
              </Element>
            </div>
          </styled.div>
          <styled.div>
            <div
              className="c6 c7"
            >
              <Element>
                <TypografiBase
                  type="element"
                >
                  <p
                    className="typo-element"
                  >
                    buc:form-actions
                  </p>
                </TypografiBase>
              </Element>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </Component>
  </Se>
</SEDPanelHeader>
`;
